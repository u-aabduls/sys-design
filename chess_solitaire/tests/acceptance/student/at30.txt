-- ACCEPTANCE TEST #30: VIOLATION CASE: moves - is_possible_move

-- Setup some chess pieces for the game
setup_chess (K,1,1)
setup_chess (Q,1,2)
setup_chess (K,1,3)
setup_chess (B,1,4)
setup_chess (R,2,1)
setup_chess (P,2,2)
setup_chess (B,2,3)
setup_chess (R,2,4)
setup_chess (N,3,1)
setup_chess (Q,3,2)
setup_chess (P,3,3)
setup_chess (Q,3,4)
setup_chess (B,4,1)
setup_chess (B,4,2)
setup_chess (R,4,3)
setup_chess (K,4,4)

-- attempt to move and capture before game has started
move_and_capture(1,-2,2,3)

-- start game
start_game

-- attempt to move and capture from an invalid slot_1
move_and_capture(1,-2,2,3)

-- attempt to move and capture from an invalid slot_2
move_and_capture(1,2,2,-3)

-- attempt to move and capture from an unoccupied slot_1
move_and_capture(1,2,2,3)

-- attempt to move and capture to an unoccupied slot_2
move_and_capture(1,3,2,3)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(3,2,2,4)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(3,2,4,4)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(1,4,2,4)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(3,1,2,4)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(2,2,3,1)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(3,3,4,4)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(3,4,2,2)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(2,3,4,1)

-- attempt to move and capture to a position not reachable
-- by the chess piece at (r1,c1)
move_and_capture(4,2,2,2)