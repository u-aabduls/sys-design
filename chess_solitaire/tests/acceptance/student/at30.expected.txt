  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(K,1,1)
  # of chess pieces on board: 1
  Game being Setup...
  K...
  ....
  ....
  ....
->setup_chess(Q,1,2)
  # of chess pieces on board: 2
  Game being Setup...
  KQ..
  ....
  ....
  ....
->setup_chess(K,1,3)
  # of chess pieces on board: 3
  Game being Setup...
  KQK.
  ....
  ....
  ....
->setup_chess(B,1,4)
  # of chess pieces on board: 4
  Game being Setup...
  KQKB
  ....
  ....
  ....
->setup_chess(R,2,1)
  # of chess pieces on board: 5
  Game being Setup...
  KQKB
  R...
  ....
  ....
->setup_chess(P,2,2)
  # of chess pieces on board: 6
  Game being Setup...
  KQKB
  RP..
  ....
  ....
->setup_chess(B,2,3)
  # of chess pieces on board: 7
  Game being Setup...
  KQKB
  RPB.
  ....
  ....
->setup_chess(R,2,4)
  # of chess pieces on board: 8
  Game being Setup...
  KQKB
  RPBR
  ....
  ....
->setup_chess(N,3,1)
  # of chess pieces on board: 9
  Game being Setup...
  KQKB
  RPBR
  N...
  ....
->setup_chess(Q,3,2)
  # of chess pieces on board: 10
  Game being Setup...
  KQKB
  RPBR
  NQ..
  ....
->setup_chess(P,3,3)
  # of chess pieces on board: 11
  Game being Setup...
  KQKB
  RPBR
  NQP.
  ....
->setup_chess(Q,3,4)
  # of chess pieces on board: 12
  Game being Setup...
  KQKB
  RPBR
  NQPQ
  ....
->setup_chess(B,4,1)
  # of chess pieces on board: 13
  Game being Setup...
  KQKB
  RPBR
  NQPQ
  B...
->setup_chess(B,4,2)
  # of chess pieces on board: 14
  Game being Setup...
  KQKB
  RPBR
  NQPQ
  BB..
->setup_chess(R,4,3)
  # of chess pieces on board: 15
  Game being Setup...
  KQKB
  RPBR
  NQPQ
  BBR.
->setup_chess(K,4,4)
  # of chess pieces on board: 16
  Game being Setup...
  KQKB
  RPBR
  NQPQ
  BBRK
->move_and_capture(1,-2,2,3)
  # of chess pieces on board: 16
  Error: Game not yet started
  KQKB
  RPBR
  NQPQ
  BBRK
->start_game
  # of chess pieces on board: 16
  Game In Progress...
  KQKB
  RPBR
  NQPQ
  BBRK
->move_and_capture(1,-2,2,3)
  # of chess pieces on board: 16
  Error: (1, -2) not a valid slot
  KQKB
  RPBR
  NQPQ
  BBRK
->move_and_capture(1,2,2,-3)
  # of chess pieces on board: 16
  Error: (2, -3) not a valid slot
  KQKB
  RPBR
  NQPQ
  BBRK
->move_and_capture(1,2,2,3)
  # of chess pieces on board: 15
  Game In Progress...
  K.KB
  RPQR
  NQPQ
  BBRK
->move_and_capture(1,3,2,3)
  # of chess pieces on board: 14
  Game In Progress...
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(3,2,2,4)
  # of chess pieces on board: 14
  Error: Invalid move of Q from (3, 2) to (2, 4)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(3,2,4,4)
  # of chess pieces on board: 14
  Error: Invalid move of Q from (3, 2) to (4, 4)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(1,4,2,4)
  # of chess pieces on board: 14
  Error: Invalid move of B from (1, 4) to (2, 4)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(3,1,2,4)
  # of chess pieces on board: 14
  Error: Invalid move of N from (3, 1) to (2, 4)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(2,2,3,1)
  # of chess pieces on board: 14
  Error: Invalid move of P from (2, 2) to (3, 1)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(3,3,4,4)
  # of chess pieces on board: 14
  Error: Invalid move of P from (3, 3) to (4, 4)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(3,4,2,2)
  # of chess pieces on board: 14
  Error: Invalid move of Q from (3, 4) to (2, 2)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(2,3,4,1)
  # of chess pieces on board: 14
  Error: Invalid move of K from (2, 3) to (4, 1)
  K..B
  RPKR
  NQPQ
  BBRK
->move_and_capture(4,2,2,2)
  # of chess pieces on board: 14
  Error: Invalid move of B from (4, 2) to (2, 2)
  K..B
  RPKR
  NQPQ
  BBRK
